packages/rx_bloc_cli/mason_templates/bricks/rx_bloc_base/__brick__/lib/base/common_blocs/user_account_bloc.dart

```diff
@@ -0,0 +1,109 @@
+import '../common_use_cases/logout_use_case.dart';
+import '../utils/validators/validators.dart';
+
+part 'user_account_bloc.rxb.g.dart';
+part 'user_account_bloc_extensions.dart';
+
+// ignore_for_file: avoid_types_on_closure_parameters
+
+abstract class UserAccountBlocEvents {
+  @RxBlocEvent(type: RxBlocEventType.behaviour, seed: '')
+  void setUsername(String username);
+
+  @RxBlocEvent(type: RxBlocEventType.behaviour, seed: '')
+  void setPassword(String password);
+
+  void login();
+
+  void logout();
+}
+
+abstract class UserAccountBlocStates {
+  /// State indicating whether the user is logged in successfully
+  Stream<bool> get loggedIn;
+
+  /// The currently entered username state
+  Stream<String> get username;
+
+  /// The currently entered password state
+  Stream<String> get password;
+
+  /// The state indicating whether we show errors to the user
+  Stream<bool> get showErrors;
+
+  /// The loading state
+  Stream<bool> get isLoading;
+
+  /// The error state
+  Stream<String> get errors;
+}
+
+@RxBloc()
+class UserAccountBloc extends $UserAccountBloc {
+  UserAccountBloc(
+    this._loginUseCase,
+    this._logoutUseCase, {
+    this.fieldValidators = const LoginFieldValidators(),
+  });
+
+  final LoginFieldValidators fieldValidators;
+  final LoginUseCase _loginUseCase;
+  final LogoutUseCase _logoutUseCase;
+
+  @override
+  Stream<bool> _mapToLoggedInState() => Rx.merge([
+        _$logoutEvent
+            .switchMap((value) {
+              _resetCredentials();
+              return _logoutUseCase.execute().asResultStream();
+            })
+            .whereSuccess()
+            .map((_) => false),
+        _$loginEvent
+            .validateCredentials(this)
+            .switchMap((args) => _loginUseCase
+                .execute(username: args.username, password: args.password)
+                .asResultStream())
+            .whereSuccess(),
+      ]).startWith(false).share();
+
+  @override
+  Stream<String> _mapToUsernameState() => _$setUsernameEvent
+      .validateField(fieldValidators.validateEmail)
+      .shareReplay(maxSize: 1);
+
+  @override
+  Stream<String> _mapToPasswordState() => Rx.merge([
+        _$setPasswordEvent.validateField(fieldValidators.validatePassword),
+        errorState.map(
+          (event) => throw RxFieldException(
+            error: 'Wrong email or password',
+            fieldValue: _$setPasswordEvent.value,
+          ),
+        )
+      ]).shareReplay(maxSize: 1);
+
+  @override
+  Stream<String> _mapToErrorsState() =>
+      errorState.map((error) => error.toString());
+
+  @override
+  Stream<bool> _mapToIsLoadingState() => loadingState;
+
+  @override
+  Stream<bool> _mapToShowErrorsState() => _$loginEvent
+      .switchMap((event) => _validateAllFields().map((event) => true))
+      .startWith(false)
+      .shareReplay(maxSize: 1);
+}
+
+class _LoginCredentials {
+  _LoginCredentials(this.username, this.password);
+  final String username;
```

Comment:
Since the `user_account_bloc_extensions` are a `part of` the `user_account_bloc` file, the mentioned class won't be accessible outside. The files containing the `part` and `part of` keywords can be understood as a single file, but split into multiple ones because of readability.

Because the `_LoginCredentials` class is used only in some places inside the `UserAccountBloc`, there is not really a need for exposing it to the rest of the app.